R/run.R

```diff
@@ -684,15 +716,25 @@ CmdStanMCMCProcs <- R6::R6Class(
           }
           if (regexpr("seconds (Total)", line, fixed = TRUE) > 0) {
             private$proc_total_time_[[id]] <- as.double(trimws(sub("seconds (Total)", "", line, fixed = TRUE)))
+            next_state <- 5
+            state <- 5
           }
           if (regexpr("seconds (Sampling)", line, fixed = TRUE) > 0) {
-            
             private$proc_section_time_[id, "sampling"] <- as.double(trimws(sub("seconds (Sampling)", "", line, fixed = TRUE)))
+            next_state <- 5
+            state <- 5
           }
           if (regexpr("seconds (Warm-up)", line, fixed = TRUE) > 0) {
             private$proc_section_time_[id, "warmup"] <- as.double(trimws(sub("Elapsed Time: ", "", sub("seconds (Warm-up)", "", line, fixed = TRUE), fixed = TRUE)))
+            next_state <- 5
+            state <- 5
           }
-          if (state > 1 && state < 5) {
+          if (regexpr("Gradient evaluation took",line, fixed = TRUE) > 0
+              || regexpr("leapfrog steps per transition would take",line, fixed = TRUE) > 0
+              || regexpr("Adjust your expectations accordingly!",line, fixed = TRUE) > 0) {
+            ignore_line <- TRUE
```

Comment:
We now print everything after metadata. I decided to mask the above lines that I am not sure are very helpful messages and mostly just spam the cmdline.